test/controllers/offer.controller.test.js

```diff
@@ -1444,297 +841,823 @@ describe('Offer Controller', () => {
           });
         });
       });
+    });
+  });
 
-      context('with unauthorized user access token', () => {
-        it('returns an error', (done) => {
+  describe('Get all offers', () => {
+    const endpoint = '/api/v1/offer/all';
+    const method = 'get';
+    let user2Token;
+    let vendor2Token;
+
+    const user2 = UserFactory.build(
+      { role: USER_ROLE.USER, activated: true },
+      { generateId: true },
+    );
+
+    const vendor2 = UserFactory.build(
+      { role: USER_ROLE.VENDOR, activated: true },
+      { generateId: true },
+    );
+
+    const editorUser = UserFactory.build({ role: USER_ROLE.EDITOR, activated: true });
+
+    const userProperties = PropertyFactory.buildList(
+      18,
+      { addedBy: vendorUser._id, updatedBy: vendorUser._id },
+      { generateId: true },
+    );
+
+    const user2Properties = PropertyFactory.buildList(
+      8,
+      { addedBy: vendor2._id, updatedBy: vendor2._id },
+      { generateId: true },
+    );
+
+    const userEnquiries = userProperties.map((_, index) =>
+      EnquiryFactory.build(
+        {
+          propertyId: userProperties[index]._id,
+          userId: regularUser._id,
+        },
+        { generateId: true },
+      ),
+    );
+
+    const user2Enquiries = user2Properties.map((_, index) =>
+      EnquiryFactory.build(
+        {
+          propertyId: user2Properties[index]._id,
+          userId: user2._id,
+        },
+        { generateId: true },
+      ),
+    );
+
+    const userOffers = userProperties.map((_, index) =>
+      OfferFactory.build(
+        {
+          propertyId: userProperties[index]._id,
+          enquiryId: userEnquiries[index]._id,
+          userId: regularUser._id,
+          vendorId: vendorUser._id,
+          referenceCode: '123456XXX',
+        },
+        { generateId: true },
+      ),
+    );
+
+    const user2Offers = user2Properties.map((_, index) =>
+      OfferFactory.build(
+        {
+          propertyId: user2Properties[index]._id,
+          enquiryId: user2Enquiries[index]._id,
+          userId: user2._id,
+          vendorId: vendor2._id,
+          referenceCode: '123456XXX',
+        },
+        { generateId: true },
+      ),
+    );
+
+    beforeEach(async () => {
+      adminToken = await addUser(adminUser);
+      userToken = await addUser(regularUser);
+      vendorToken = await addUser(vendorUser);
+      user2Token = await addUser(user2);
+      vendor2Token = await addUser(vendor2);
+    });
+
+    itReturnsEmptyValuesWhenNoItemExistInDatabase({
+      endpoint,
+      method,
+      user: regularUser,
+      useExistingUser: true,
+    });
+
+    itReturnsEmptyValuesWhenNoItemExistInDatabase({
+      endpoint,
+      method,
+      user: adminUser,
+      useExistingUser: true,
+    });
+
+    itReturnsEmptyValuesWhenNoItemExistInDatabase({
+      endpoint,
+      method,
+      user: vendorUser,
+      useExistingUser: true,
+    });
+
+    describe('when offers exist in db', () => {
```

Comment:
Lets change this to context since you started the sentence with when

```suggestion
    context('when offers exist in db', () => {
```